src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/writers/GeoJSONWriter.java

```diff
@@ -88,91 +79,83 @@ public void endTemplateOutput() throws IOException {
     public void writeLink(String href, String rel, String mimeType, String title, String method)
             throws IOException {
         if (href != null) {
-            startObject();
+            writeStartObject();
             if (title != null) {
-                writeFieldName("title");
+                generator.writeFieldName("title");
                 writeValue(title);
             }
             if (mimeType != null) {
-                writeFieldName("type");
+                generator.writeFieldName("type");
                 writeValue(mimeType);
             }
             if (rel != null) {
-                writeFieldName("rel");
+                generator.writeFieldName("rel");
                 writeValue(rel);
             }
             if (method != null) {
-                writeFieldName("method");
+                generator.writeFieldName("method");
                 writeValue(method);
             }
-            writeFieldName("href");
+            generator.writeFieldName("href");
             writeValue(href);
-            endObject();
+            writeEndObject();
         }
     }
 
     public void writePagingLinks(String mimeType, String previous, String next) throws IOException {
 
-        writeFieldName("links");
-        startArray();
+        generator.writeFieldName("links");
+        writeStartArray();
```

Comment:
The startArray has an EncodingHint parameter since is defined from the TemplateWriter class.writeStartArray() is available only for json writer as is it is the same method of the JsonGenerator. I switched to that one to avoid to invoke startArray with null encoding hints.